### DISCUSSION TITLE
How to add vite plugin?

### DISCUSSION THREAD

**Discussion by iodum** [vuepress/core#197] [Q&A] [answered]
I want to import SVG as a component and use vite-svg-loader plugin for this in Vite+Vue3. It works correctly
But when I try to add this plugin to vuepress config like 
```js
module.exports = {
  bundler: '@vuepress/vite',
  bundlerConfig: {
    viteOptions: {
      plugins: [svgLoader()]
    }
  },
  ...
````
  I get error
  `Failed to execute 'createElement' on 'Document': The tag name provided ('...svg') is not a valid name.`.
  
  How can I use the plugin with Vuepress or is there another way to solve the SVG problem?

**Comment by meteorlxy** [comment, marked as answer]
I should work. Please provide a reproduce repo.

**Reply by iodum** [reply]
https://github.com/iodum/vuepress-vite

**Reply by meteorlxy** [reply]
Should be an issue of `vite-svg-loader` because it does not provide a default export.

We didn't enable `esModuleInterop` when transforming ts code.

Use following snippet instead:

```js
// .vuepress/config.js
const svgLoader = require('vite-svg-loader')

module.exports = {
  bundler: '@vuepress/vite',
  bundlerConfig: {
    viteOptions: {
      plugins: [svgLoader()]
    }
  },
  ...
```

**Reply by iodum** [reply]
THis snippet doesn't work too. Ok, I'll try to find another svg plugin

**Comment by plexus77** [comment]
You can achieve what you want by doing the following:

```import svgTest from './../public/icons/test.svg?raw'```

No loader needed!  Output it using `v-html`

**Comment by ptandler** [comment]
I got a suggestion from @Mister-Hope that helped me:
https://github.com/vuepress/vuepress-next/issues/795#issuecomment-1090461964
